### ISSUE TITLE
How to maintain socket connection in background??

### ISSUE THREAD

**Issue by vikrantrajput** [pkyeck/socket.IO-objc#232]
Hello everyone, I am using socket.IO-objc in my project using XCode9.0. I am trying to paring or connect with hardware device with using particular IP Address and Port which connect successfully. 
 I am using below code. ` // create socket.io client instance
    socketIO = [[SocketIO alloc] initWithDelegate:self];
[socketIO connectToHost:@"myHost" onPort:port];`

and then I am getting response from 

- (void) connection:(NSURLConnection *)connection didReceiveData:(NSData *)data  in SocketIO.m class. Now data coming in every seconds. It is working well. But when I lock my iPhone then connection getting break. This is the main issue. So how I can use socket connection in background. Please guide me I stuck from last many days.

Thanks

**Comment by mhisham** [comment]
Yes, ios break all socket connections when your app goes in background, for that reason you have to support Push notification for important actions or get the updates every time your app reconnect again. Hope that helps
